packet_entity.go

```diff
@@ -27,16 +38,31 @@ func (pe *PacketEntity) Fetch(key string) (interface{}, bool) {
 	return v, ok
 }
 
+// A function that can handle a game event.
+type packetEntityHandler func(*PacketEntity, EntityEventType) error
+
+// Registers a packet entity event handler.
+func (p *Parser) OnPacketEntity(fn packetEntityHandler) {
+	p.packetEntityHandlers = append(p.packetEntityHandlers, fn)
+}
+
 // Internal callback for CSVCMsg_PacketEntities.
 func (p *Parser) onCSVCMsg_PacketEntities(m *dota.CSVCMsg_PacketEntities) error {
+	// Skip processing if we're configured not to.
 	if !p.ProcessPacketEntities {
 		return nil
 	}
 
 	_debugfl(5, "pTick=%d isDelta=%v deltaFrom=%d updatedEntries=%d maxEntries=%d baseline=%d updateBaseline=%v", p.Tick, m.GetIsDelta(), m.GetDeltaFrom(), m.GetUpdatedEntries(), m.GetMaxEntries(), m.GetBaseline(), m.GetUpdateBaseline())
 
+	// Skip processing full updates after the first. We'll process deltas instead.
```

Comment:
I think it's fine to leave it in for now but keep it in the back of our minds if any bugs come up. Assuming that replays are recorded on the server, we should have any 'dropped' packets.

I think there are other areas for performance optimisations.